modules/Bio/EnsEMBL/Transcript.pm

```diff
@@ -2161,21 +2173,20 @@ sub translate {
                                -alphabet => 'dna',
                                -id       => $display_id );
 
-# Signature for BioPerl 1.6 should be (https://metacpan.org/pod/Bio::PrimarySeqI)
-#     my ($terminator, $unknown, $frame, $codonTableId, $complete,
-#       $complete_codons, $throw, $codonTable, $orf, $start_codon, $offset);
-# 
-# Temporarily hardcoding the two parameters below.
-# possibly to drive later on using *_start_NF or *_end_NF transcript attributes
-
+  # Signature for BioPerl 1.6 should be (https://metacpan.org/pod/Bio::PrimarySeqI)
+  #     my ($terminator, $unknown, $frame, $codonTableId, $complete,
+  #       $complete_codons, $throw, $codonTable, $orf, $start_codon, $offset);
+ 
   my $cds_complete = undef;
   my $codons_complete = undef;
-
-  my $translation = $peptide->translate( -terminator => undef, -unknown => undef, -frame => undef, 
-                         -codonTable_Id => $codon_table_id,
-                         -complete => $cds_complete, -complete_codons => $codons_complete,
-                         -throw => undef, -codonTable => undef,
-                         -orf => undef, -start => undef, -offset => undef );
+  $cds_complete = 0 if ( $mrna5_nf + $cds5_nf + $mrna3_nf + $cds3_nf > 0);
```

Comment:
Imho,  only `cds` related flags should be checked ($cds5_nf + $cds3_nf).

Reply:
Hi @vsitnik - out of curiosity, if the CDS start (or end) is not found, would it not be the case that the mRNA start (or end) is also not found? I was wondering if it could be a useful check to have both CDS and mRNA attribute information, and also if there might be a case where the mRNA_*_NF attribute could be flagged as incomplete but the CDS-based attribute might be missing.

Reply:
@nwillhoft , as far as I understand it, the CDS is just a substring of mRNA. I suspect you can have partial mRNA but intact CDS. But for translation (peptide sequence inference) we need just CDS in this case.

Reply:
Thanks @vsitnik